### PR TITLE
Add wix installer toolset in appveyor setup

### PR DESCRIPTION
**Pull request by mwichmann**
The msi packaging tests won't be run if this is not installed.
These is no change at all to scons itself - so no doc or test, and no change to CHANGES.txt.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [X] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -6,10 +6,12 @@ install:
   - python --version
   - pip install pypiwin32
   - set STATIC_DEPS=true & pip install lxml
+  
   - choco install --allow-empty-checksums dmd
   - choco install --allow-empty-checksums ldc
   - choco install --allow-empty-checksums swig
   - choco install --allow-empty-checksums vswhere  
+  - choco install --allow-empty-checksums vixtoolset
```
Comment (bdbaddog) [inline .appveyor.yml]: Should be wixtoolset

Comment (mwichmann) [reply .appveyor.yml]: aaaargh

**Comment by mwichmann** [comment]
Hmmm, it installs fine but apparently isn't being found, as the tests are still skipped. I had this working after manual installation, does chocolatey not add the path?

```
139/294 (47.28%) C:\\Python27\\python.exe -tt test\packaging\msi\explicit-target.py
No 'candle' found; skipping test
NO RESULT for test of C:\projects\scons\src\script\scons.py
	at line 46 of test\packaging\msi\explicit-target.py
140/294 (47.62%) C:\\Python27\\python.exe -tt test\packaging\msi\file-placement.py
No 'candle' found; skipping test
NO RESULT for test of C:\projects\scons\src\script\scons.py
	at line 45 of test\packaging\msi\file-placement.py
141/294 (47.96%) C:\\Python27\\python.exe -tt test\packaging\msi\package.py
No 'candle' found; skipping test
NO RESULT for test of C:\projects\scons\src\script\scons.py
	at line 45 of test\packaging\msi\package.py

```

**Comment by mwichmann** [comment]
expecting is to have landed in `c:\Program Files (x86)\WiX Toolset 3.11\bin`

**Comment by bdbaddog** [comment]
Take a look at what I did with clang and java for windows. Something similar is likely right, and/or we add chocolatey's path to the default windows path..  That might be the simplest.. Try that?

**Comment by mwichmann** [comment]
No, the choco path won't help, that's mainly its own internal stuff.  If the thing it's installing is a standard pkg, it just goes the normal place, for wix it's the one a few comments above in the 32-bit Program FIles.  If it's not a full installer thing, it goes into whatever is the ChocolateyToolsLocation path, but even there you get subdirs, keeping the helpful Windows model of everything-gets-their-very-own-directory.  All this from experimentation the other day, I'm no choco expert.  I had to learn about ChocolateyToolsLocation because it defaults to C:\tools, which an anti-ransomware tool I have objects to and end ends up blocking - things can't write to the "root directory'.

**Comment by bdbaddog** [comment]
I have a blurb of logic to  check reasonable default paths for wix install and add to PATH.
See:
https://github.com/bdbaddog/scons/commit/a30df639b15c681e1b0b8585b6982b4ff929a2bd

All the msi tests then fail with undefined X_MSI_LANGUAGE.  Care to chase it down?

**Comment by mwichmann** [comment]
Hmmm, choco does make shims for _some_ of the things it installs, I think only for the ones that don't have a classic installer and end up in Program Files* ... so that doesn't include wix.

**Comment by bdbaddog** [comment]
I added a note with a pointer to a commit on a branch in my repo which
fixes finding the tool, but then yields some errors in the logic.

Care to take on resolving that and pulling in my location finding logic?

**Comment by mwichmann** [comment]
Hmmm, magical new failure, about multiple environments. e.g. the file-placement test. I don't recall seeing this when I was creating the patch. Guess this will need more work.
```
File "C:\Users\appveyor\AppData\Local\Temp\1\testcmd.4516.toeggl23\SConstruct", line 19, in <module>
140/294 (47.62%) C:\\Python36\\python.exe -tt test\packaging\msi\file-placement.py
FAILED test of C:\projects\scons\src\script\scons.py
	at line 624 of C:\projects\scons\testing\framework\TestCommon.py (_complete)
	from line 727 of C:\projects\scons\testing\framework\TestCommon.py (run)
	from line 393 of C:\projects\scons\testing\framework\TestSCons.py (run)
	from line 67 of test\packaging\msi\file-placement.py
C:\projects\scons\src\script\scons.py returned 2
STDOUT =========================================================================
scons: Reading SConscript files ...
STDERR =========================================================================
scons: *** Two environments with different actions were specified for the same target: foo-1.2
File "C:\Users\appveyor\AppData\Local\Temp\1\testcmd.4980.bbqbmgk_\SConstruct", line 12, in <module>
```

**Comment by mwichmann** [comment]
So I still don't get this: all three msi tests fail with the same kind of error, which claims multiple environments. I see only one environment being created... this is the one from `explicit-target.py`. 
```
scons: *** Two environments with different actions were specified for the same target: mypackage.msi
```

**Comment by mwichmann** [comment]
So this should be finding candle and light *without* adding the toolset with choco, since we hear the Appveyor images already come with it.  Should I remove the .appveyor.yml bit and see if it does find it?  I have installed the standard package and it sets the WIX environment varialble, which the patch now fetches; and also the last-chance location should be the correct one.

**Comment by dmoody256** [comment]
Appveyor responded this:
http://help.appveyor.com/discussions/questions/34138-where-is-the-wix-binaries-located

**Comment by mwichmann** [comment]
> It doesn't. they're not all in choco's path. I've gone down this route before trust me on this...
> […](#)

In my test VM it does find it with the code in this PR, I just tried - and the location is the same as we just heard back from appveyor.

**Comment by mwichmann** [comment]
Closing this, will proceed with #3294 instead.

**Pull request closed without merging**